docs/sandboxing/README.md

```diff
@@ -10,9 +10,95 @@ There is no clear solution to this yet, but it should take heavy inspiration fro
+- `--dev /dev --dev-bind-try /dev/dri /dev/dri` - Allow access to DRM devices for hardware accelerated rendering
+- `--proc /proc` - Mount namespaced procfs at /proc
+- `--unsetenv DBUS_SESSION_BUS_ADDRESS` - The sandbox doesn't have access to the dbus daemon anyway
+- Read-only bind mounts
+  - `/etc/drirc`
+  - `/etc/gai.conf`
+  - `/etc/gnutls`
+  - `/etc/hostname`
+  - `/etc/hosts`
+  - `/etc/je_malloc.conf`
+  - `/etc/localtime`
+  - `/etc/machine-id` - TODO: what are consequences of omitting this
+  - `/etc/os-release` - TODO: what are consequences of omitting this
+  - `/etc/resolv.conf`
+  - `/etc/selinux`
+  - `/etc/timezone`
+  - `/usr`
+  - `/bin`
+  - `/sbin`
+  - `/lib`
+  - `/lib32`
+  - `/lib64`
+  - `/sys/class`
+  - `/sys/dev/char`
+  - `/sys/devices/pci????:??` - Possibly multiple?
+  - `/sys/devices/system/cpu`
+  - Path to game assets and libraries
+- Read-write bind mounts
+  - Path to game root (`.minecraft`)
+
+##### Desktop integration
+
+- Read-only bind mounts
+  - `$XDG_RUNTIME_DIR/pulse` - Access to PulseAudio daemon
+  - `$XDG_RUNTIME_DIR/pipewire-?` - Access to PipeWire daemon (`pipewire-0` if unspecified)
+  - `$XAUTHORITY` - X11 access
+  - `/tmp/.X11-unix/X?` - X11 access (`X0` if `DESKTOP=:0`)
+  - `$XDG_RUNTIME_DIR/$WAYLAND_DISPLAY` - Wayland access (`$WAYLAND_DISPLAY` can also be an absolute path)
+
+##### Distro-specific configuration
+
+To solve issues with distro-specific symlinks of files, mostly in `/etc` (i.e. `resolv.conf -> /run/systemd/resolve/stub-resolv.conf`), bind mounts SHOULD use canonical file paths for the host side, while preserving the original path in the sandbox.
+
+- Read-only bind mounts
+  - `/nix/store` - Required for Nix installations
+  - `/nix/var/nix/profiles` - Required for Nix installations
+  - `/run/current-system/sw` - Required for NixOS installations
+
+##### Supporting `xdg-open`
+
+By default xdg-open would launch applications inside our sandbox. We can utilize `xdg-desktop-portal` to open links/paths on the host system.
+To do this we *need* dbus, but we neither ensure that it's bound inside the sandboxing, neither do we pass the `DBUS_SESSION_BUS_ADDRESS` variable to the sandbox.
+We don't want to allow full access to dbus, as that could allow sandbox escapes (some endpoints allow executing arbitrary commands on the host).
+The Flatpak project maintains a tool called `xdg-dbus-proxy` that acts as a filtering reverse proxy for dbus.
+
+As we only need access to `org.freedesktop.portal.*` the call to xdg-dbus-proxy is quite simple:
+
+```console
+# PROXIED_BUS_PATH can be an arbitrary path on the host
+$ xdg-dbus-proxy $DBUS_SESSION_BUS_ADDRESS $PROXIED_BUS_PATH --filter '--call=org.freedesktop.portal.*=*' '--broadcast=org.freedesktop.portal.*=@/org/freedesktop/portal/*'
+```
+
+To give the sandbox access to this filtered dbus, the `bwrap` call must be extended by the following two parameters:
+- `--bind $PROXIED_BUS_PATH /tmp/bus` - TODO: can this be read-only?
```

Comment:
I am fairly certain DBUS can be read only, most events are broadcasts, and minecraft shouldn't be needing RMI

Reply:
I am not sure how read-only sockets are supposed to work in the first place. You HAVE to be able to write to communicate at all, but I already have read-only sockets above. So maybe RO has no meaning for sockets?

Reply:
That's not how DBus or sockets work at all. If it's read only, you can't connect, it's that simple.
If you were only unable to send, the DBus protocol would still be unusable (plus "event broadcast" isn't why this is a thing)

Reply:
In that case I am unsure of how all the other sockets work in my example implementation, as X11, Wayland, Pulse and PipeWire are all mounted using `--ro-bind-try`

Reply:
(while Wayland, Pulse and PipeWire could go unused, at the very least X11 should also not work in that case)

Reply:
Well, unlike wayland, X11 does support TCP sockets, so that'd be covered by sharing the network namespace.
Also what did you even test? Only the launcher?

Reply:
I am using these flags to launch the game (per-instance sandboxing). I can launch and play the game without any hiccups using my bwrap wrapper in Prism Launcher